### PR TITLE
Update q-connection.js

### PR DESCRIPTION
**Pull request by zarutian**
Not yet tested. Should solve this issue:

Side A has remoteB to localB on Side B.
Side A passes remoteB in send message to something on side B.
Side B makes a remote call it remoteBA to sides A remoteB when it should just use localB.

### CODE REVIEW

File: q-connection.js [hunk 1]
```diff
@@ -215,6 +216,8 @@ function Connection(connection, local, options) {
             }));
             return response;
         });
+        remotes.set(r,id);
+        return r;
```
Comment (kriskowal) [inline q-connection.js]: Use full word variable names, e.g., remote.

**Comment by kriskowal** [comment]
Good idea to split the reference domain for locals and remotes. Get some passing tests and merge.

Would be good for us to step up the professionalism and use nyc or istanbul to get the test coverage up to 100%.

**Comment by zarutian** [comment]
Travis CI build passes.
Removed node_js versions 0.4 and 0.7 as both chocked on stuff unrelated to this codebase afaisi.
Will Merge this pull request in one day if no objection is recived.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: node_js
 node_js:
-  - "0.4"
   - "0.6"
-  - "0.7"
   - "0.8"
```
Comment (kriskowal) [inline .travis.yml]: We can live with just 0.10 and 4. 5 is good for catching node bugs early, but is not Long-Term-Support track.

**Comment by kriskowal** [comment]
:ok:

**Comment by zarutian** [comment]
Merging.

**Pull request merged**